### DISCUSSION TITLE
ScriptReference/AnimationClip.SetCurve

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#619] [Notes]
# ScriptReference/AnimationClip.SetCurve

https://docs.unity3d.com/ScriptReference/AnimationClip.SetCurve.html

**Comment by forestrf** [comment]
Use `AnimationUtility.SetEditorCurve` in the editor, this function always interpolates using euler and does not preserve the interpolation mode (Quaternion or Euler) which is selected by checking if propertyName is `localEulerAngles`, `localEulerAnglesRaw` or `localEulerAnglesBaked` which represent `Quaternion`, `Euler Angles (Quaternion)` and `Euler Angles` respectively.
